chips/sam4l/src/usart.rs

```diff
@@ -117,7 +117,8 @@ impl USART {
     }
 
     fn set_baud_rate(&self, baud_rate: u32) {
-        let cd = 48000000 / (8 * baud_rate);
+        let system_frequency = 48000000;
```

Comment:
I guess ideally we'd want to get this value dynamically? Can we do it by sort of brute force--just looking at the main clock register and prescalar and computing it based on what we know the frequencies to be? It might be a bit expensive, but presumably we don't make that call often (most likely just once on setup, or at worse when a baud rate is changed)